### PR TITLE
Allow rotation of axes tick labels

### PR DESCRIPTION
**Pull request by mlq**
This patch allows arbitrary rotation of tick labels. If the rotation is the same for every label on the given axes, it will apply a global `xticklabel style`. If the rotation values differs, it will create an array of the rotation value and apply the one at the given tick index.

In addition, I've attached a small test set with different rotated labels on both, the x and y axes.

### CODE REVIEW

File: matplotlib2tikz/axes.py [hunk 1]
```diff
@@ -198,6 +198,37 @@ def __init__(self, data, obj):
                     assert direction == 'inout'
                     self.axis_options.append('tick align=center')
 
+        # Set each rotation for every label
+        x_tick_labels_rotation = \
+            [label.get_rotation() for label in obj.xaxis.get_majorticklabels()]
+        if any(x_tick_labels_rotation) != 0:
+            # check if every value is the same
+            if all(x == x_tick_labels_rotation[0] for
```
Comment (nschloe) [inline matplotlib2tikz/axes.py]: I guess you can go `all(x_tick_labels_rotation = x_tick_labels_rotation[0)` here.

Comment (mlq) [inline matplotlib2tikz/axes.py]: I am sorry, but I don't think that this is working. However, I've replaced it with `len(set(x_tick_labels_rotation)) == 1`

**Comment by nschloe** [comment]
Looks good.

One possible improvement would be to make the rotation code a little function (since it's virtually the same for `x` and `y`). Just pass in a string (`x` or `y`), the label rotations, and return the code for the rotations. Append that to `self.axis_options`.

**Comment by mlq** [comment]
I've updated the patch to support more options: Now the horizontal alignment is respected (an additional parameter has to be passed to matplotlib2tikz.save in the extra dictionary) and the visibility on the axis of the ticks can be defined.

**Comment by nschloe** [comment]
Very nice work on the tests.

I see that some of the older tests needed a change in the phash, so things do look differently now. Do you know what exactly changed there? (Perhaps two screenshots would be useful.)

**Comment by mlq** [comment]
I honestly can't see any visual changes to it (also not with a tool). However, I haven't checked how phash works, but maybe it is because I add the used alignment in the generated tex file.

**Comment by nschloe** [comment]
It's a _perceptual hash_ (hence the name), so it only works on the output file. For comparing the images, you could check out http://stackoverflow.com/a/11362152/353337. (That's what I use.)

**Comment by mlq** [comment]
Okay, I did a mistake this afternoon and used the wrong method to compare the files. The difference is that before my patch, the generated tikz file shows the ticks on all axes (x: top, bottom, y: left, right), while now only those that are set by matplotlib are displayed:

Before:
Before

After:
After

**Comment by nschloe** [comment]
Ah, makes sense that so many test cases are affected then. Very nice work!

**Pull request merged**